feat(root): add GitHub Actions workflow for releasing packages

merrcury opened the pull request:
### What changed? Why was the change needed?

This fixes NV-6034

### Screenshots

<!-- If the changes are visual, include screenshots or screencasts. -->

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Related enterprise PR

<!-- A link to a dependent pull request  -->

### Special notes for your reviewer

<!-- Specific instructions or considerations you want to highlight for the reviewer. -->

</details>

merrcury committed:
feat(workflow): add GitHub Actions workflow for releasing packages

merrcury commented:
@scopsy Hey, I did some research — we’ll still need to publish the tag and release manually. I’ll put together some documentation for it. The issue with publishing the release and tag through the action is that it would tag the older version, without including this change that’s currently merged in next.

scopsy commented on .github/workflows/release-packages.yml:
```diff
@@ -0,0 +1,68 @@
+        description: 'Packages to release (comma-separated)'
+        required: true
+        type: string
+        default: '@novu/js,@novu/react,@novu/nextjs,@novu/react-native'
```
How does dependencies work? For example if next/js is dependant on js and react and js version? Will all be bumped in the same time properly?

merrcury replied:
When I run this locally this happens, it updated in all the places also. It remains `workspace:*` and as per in discussion pnpm publish take cares of that
<img width="614" alt="Screenshot 2025-06-10 at 13 56 13" src="https://github.com/user-attachments/assets/364d1f60-5e56-4f17-a41d-a5fd72f37dc8" />

scopsy reviewed (approved):
How are we going to test it out? Maybe doing a nightly release or something like this?

scopsy commented on .github/workflows/release-packages.yml:
```diff
@@ -0,0 +1,68 @@
+          node-version: '20.19.0'
+          cache: 'pnpm'
+
+      - name: Install Dependencies
+        shell: bash
+        run: pnpm install --frozen-lockfile
```
Should we maybe use the slim version of the setup-project?

merrcury replied:
We certainly can

scopsy commented:
Awesome, than it looks good on my end 💪 Let's just test it with a nightly after merged to make sure everything works as expected. Also for the nightly, does it uses npm --tag ? to make sure it's not published as the latest version

scopsy committed:
Merge branch 'next' into feat/releasepipeline

merrcury committed:
feat(workflow): add support for nightly releases in package workflow

- Introduced a new input parameter 'nightly' to the release workflow.
- Implemented logic to set a nightly version and handle nightly releases for versioning, changelog generation, and package publishing.
- Updated the publish step to tag nightly releases appropriately.
